students/arun_nalla/session03/Lesson3_Mailroom.py

```diff
@@ -0,0 +1,62 @@
+#!usr/bin/en python
+donors_list = [('Aron 123', [10,5]), ('Becky', [3,30]), ('Charlie',[10,26]),('Thomas',[22,9,4,5]),('Nancy',[1,2,3])]
+donors_name = [x for x,y in donors_list]
+donors_number = {x: len(y) for x,y in donors_list}
+donors_total = {x:sum(y) for x,y in donors_list}
+
+def thank_you(seq):
+    name_input = input("Enter a name from the list: ").title()
+    if name_input in donors_name:
+        print("\nDear Mr.{},\nThank you for donation of ${:.2f} for XYZ Charity Organisation. Its a kind initiative from"
+        " you for a nobel cause.\n\nWith Regards\n\nXYZ\n".format(name_input, donors_total[name_input]))
+    else:
+        print ('Name entered is not in the donor list')
+        thank_you(seq)
```

Comment:
Not a good use of recursion. Please rewrite this with a `while` loop.

Besides, I thought the requirement is to add this non-existent donor to the database and receive a donation amount? Not sure if your code is following the problem requirements correctly...?